tests/pyfunc/test_pyfunc_model_with_type_hints.py

```diff
@@ -1121,6 +1117,11 @@ def predict(
 
     assert mock_warning.call_count == 0
 
+    # Check import does not trigger any warning (from builtin sub-classes)
+    subprocess.check_call(
+        [sys.executable, "-W", "error::TypeHintWarning", "-c", "import mlflow.pyfunc"]
```

Comment:
```suggestion
        [sys.executable, "-W", "error::TypeHintWarning", "-c", "import mlflow.pyfunc.model"]
```

`mlflow.pyfunc` is a lazy module. Just importing it doesn't load the classes in it.